### PR TITLE
Hw2 class3 by ellie and me and hw3

### PR DESCRIPTION
**Pull request by ssinghal22**
Hi Ellie, please review. I did till Bonus round 2. TY!

### CODE REVIEW

File: tests/my_test.py [hunk 1]
```diff
@@ -10,42 +12,108 @@ def test_for_successful_response(self):
         # OR
         self.assertTrue('OK' == result.reason)
 
-
 class CareerPortalTests(unittest.TestCase):
     def setUp(self) -> None:
-        self.base_url = 'https://recruit-portnov.herokuapp.com/recruit/api/v1'
-        self.headers = {'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36'}
-
+        pass
```
Comment (EllieSky) [inline tests/my_test.py:17]: You can remove setUp if it's not needed

File: tests/my_test.py [hunk 2]
```diff
@@ -10,42 +12,108 @@ def test_for_successful_response(self):
...
+    '''
     def test_login(self):
-        positions = requests.get(self.base_url + '/positions')
+        sess = Authenticate() #sess is an object of class, creates instance of class
```
Comment (EllieSky) [inline tests/my_test.py:20]: sess is a variable pointing to an object, that is an instance of Authenticate class

File: tests/my_test.py [hunk 3]
```diff
@@ -10,42 +12,108 @@ def test_for_successful_response(self):
...
+        add_candidate = sess.add_candidate(firstName, lastName, email, password)
+        json_add_candidate = json.loads(add_candidate.text)
+        id = json_add_candidate['id']
+        print('id of added candidate is: ' + str(id))
+
+        all_candidates = sess.get_all_candidates()
+        json_all_candidates = json.loads(all_candidates.text)
+        count2 = len(json_all_candidates)
+        print('Total number of candidates now: ' + str(count2))
+        self.assertGreater(count2, count1)
```
Comment (EllieSky) [inline tests/my_test.py:82]: very nice :)

File: tests/my_test.py [hunk 4]
```diff
@@ -10,42 +12,108 @@ def test_for_successful_response(self):
...
+
+        ids = []
+        for elem in json_all_candidates:
+            ids.append(elem['id'])
+
+        if id in ids:
+            print('Candidate found with id: ' + str(id))
```
Comment (EllieSky) [inline tests/my_test.py:89]: No if... don't print, assert!!!

self.assertIn(id, ids)

File: tests/my_test.py [hunk 5]
```diff
@@ -10,42 +12,108 @@ def test_for_successful_response(self):
...
+        self.assertEqual(204, delete_candidate.status_code)
+
+        all_candidates = sess.get_all_candidates()
+        json_all_candidates = json.loads(all_candidates.text)
+
+        ids = []
+        for elem in json_all_candidates:
+            ids.append(elem['id'])
+
+        if id not in ids:
```
Comment (EllieSky) [inline tests/my_test.py:105]: self.assertNotIn(id,ids)

**Comment by ssinghal22** [comment]
Thank you!
Best, Soniya 

    On Saturday, May 30, 2020, 05:27:09 PM PDT, Ellie Yampolskaya <notifications@github.com> wrote:  
 
 


@EllieSky commented on this pull request.

In tests/my_test.py:
>  class CareerPortalTests(unittest.TestCase):
     def setUp(self) -> None:
-        self.base_url = 'https://recruit-portnov.herokuapp.com/recruit/api/v1'
-        self.headers = {'user-agent': 'Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/81.0.4044.138 Safari/537.36'}
-
+        pass

You can remove setUp if it's not needed

In tests/my_test.py:
>      def test_login(self):
-        positions = requests.get(self.base_url + '/positions')
+        sess = Authenticate() #sess is an object of class, creates instance of class

sess is a variable pointing to an object, that is an instance of Authenticate class

In tests/my_test.py:
> +       # candidate_details = {'firstName': 'Soniya', 'lastName': 'Singhal', 'email': 'ssinghal22@yahoo.com', 'password': 'abc123.', 'id': 20, 'positionsCount': 0}
+
+        firstName = 'Soniya'
+        lastName = 'Singhal'
+        email = 'ssinghal13@yahoo.com'
+        password = 'abc123.'
+        add_candidate = sess.add_candidate(firstName, lastName, email, password)
+        json_add_candidate = json.loads(add_candidate.text)
+        id = json_add_candidate['id']
+        print('id of added candidate is: ' + str(id))
+
+        all_candidates = sess.get_all_candidates()
+        json_all_candidates = json.loads(all_candidates.text)
+        count2 = len(json_all_candidates)
+        print('Total number of candidates now: ' + str(count2))
+        self.assertGreater(count2, count1)

very nice :)

In tests/my_test.py:
> +        json_add_candidate = json.loads(add_candidate.text)
+        id = json_add_candidate['id']
+        print('id of added candidate is: ' + str(id))
+
+        all_candidates = sess.get_all_candidates()
+        json_all_candidates = json.loads(all_candidates.text)
+        count2 = len(json_all_candidates)
+        print('Total number of candidates now: ' + str(count2))
+        self.assertGreater(count2, count1)
+
+        ids = []
+        for elem in json_all_candidates:
+            ids.append(elem['id'])
+
+        if id in ids:
+            print('Candidate found with id: ' + str(id))

No if... don't print, assert!!!

self.assertIn(id, ids)

In tests/my_test.py:
> +
+
+        # sess.authenticate('student@example.com', 'welcome') #login type 1
+        sess.authenticate(email, password)  #login type 2
+
+        delete_candidate = sess.delete_added_candidate(id)
+        self.assertEqual(204, delete_candidate.status_code)
+
+        all_candidates = sess.get_all_candidates()
+        json_all_candidates = json.loads(all_candidates.text)
+
+        ids = []
+        for elem in json_all_candidates:
+            ids.append(elem['id'])
+
+        if id not in ids:

self.assertNotIn(id,ids)

**Pull request closed without merging**